### PR TITLE
Modify DAIInterestRateModel.sol => DAIInterestRateModelV2.sol

### PR DESCRIPTION
**Pull request by 0age**
The primary intention behind this change is to "smooth out" the cDai interest rate model to promote greater rate stability.

It does so by increasing the `gapPerBlock`, or additional "slope" of the interest rate curve added to the Dai stability fee, from 0.05% to 2%.

There are also a few minor changes, designed to clean up the interest rate model contract.

Summary of changes:
- `gapPerBlock` constant changed from `0.05e16 / blocksPerYear` to `2e16 / blocksPerYear`
- `uint` => `uint256` (explicit > implicit)
- added parenthesis to SafeMath expressions to make ordering explicit
- used an interface (rather than an abstract contract) for the maker interfaces
- modified a tiny bit of `poke()` to add `gapPerBlock` to `stabilityFeePerBlock` _before_ subtracting `baseRatePerBlock` for resiliency against a very unlikely underflow
- named some unnamed return values

### CODE REVIEW

File: contracts/DAIInterestRateModelV2.sol [hunk 1]
```diff
@@ -0,0 +1,113 @@
+pragma solidity ^0.5.16;
+
+import "./JumpRateModel.sol";
+import "./SafeMath.sol";
+
+/**
+  * @title Compound's DAIInterestRateModel Contract (version 2)
+  * @author Compound
```
Comment (jflatow) [inline contracts/DAIInterestRateModelV2.sol:8]: (modified by Dharma)?

File: contracts/DAIInterestRateModelV2.sol [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  * rate increases.
+  */
+contract DAIInterestRateModelV2 is JumpRateModel {
+    using SafeMath for uint256;
+
+    /**
+     * @notice The additional margin per block separating the base borrow rate from the roof (2% / block).
+     * Note that this value has been increased from the original value of 0.05% per block.
+     */
+    uint256 public constant gapPerBlock = 2e16 / blocksPerYear;
```
Comment (jflatow) [inline contracts/DAIInterestRateModelV2.sol:21]: I would be a much bigger fan of this change if this were the only diff

File: contracts/DAIInterestRateModelV2.sol [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    /**
+     * @notice Calculates the current supply interest rate per block including the Dai savings rate
+     * @param cash The total amount of cash the market has
+     * @param borrows The total amount of borrows the market has outstanding
+     * @param reserves The total amnount of reserves the market has
+     * @param reserveFactorMantissa The current reserve factor the market has
+     * @return The supply rate per block (as a percentage, and scaled by 1e18)
+     */
+    function getSupplyRate(uint256 cash, uint256 borrows, uint256 reserves, uint256 reserveFactorMantissa) public view returns (uint256 supplyRateMantissa) {
```
Comment (jflatow) [inline contracts/DAIInterestRateModelV2.sol:52]: Not a huge fan of all the stylistic changes, I think the bike-shedding could be worse on-chain where the next change someone wants to make moves to a different style, and all the contracts in the protocol have different conventions and huge diffs between them

File: contracts/DAIInterestRateModelV2.sol (hunk 1 above)
Comment (hayesgm) [reply contracts/DAIInterestRateModelV2.sol:8]: On that note, even if we want to rename the contract, maybe we should keep it in the same file? It's pretty hard to see the changes directly considering git considers this a full file deletion and file addition?

Comment (0age) [reply contracts/DAIInterestRateModelV2.sol]: If I'm not mistaken, taking out all the little tweaks fixed that 👍

**Pull request merged**